payload.ts

```diff
@@ -1,5 +1,6 @@
-export type Payload = {
-  readonly method: string
+import type { RequestArguments } from './types'
+
+export type Payload = RequestArguments & {
   readonly params: readonly unknown[]
```

Comment:
again do we allow optional params here

Reply:
I'm not sure that the JSON RPC spec allows for params to not be specified, or at least every example I see of it has empty params instead of them not being there: https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_accounts

Reply:
Does say it MAY be omitted: https://www.jsonrpc.org/specification